Day0/autives/assignment1/ppm_generator.c

```diff
@@ -0,0 +1,145 @@
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+
+#ifndef max
+#define max(x, y) ((x) > (y) ? (x) : (y))
+#define min(x, y) ((x) < (y) ? (x) : (y))
+#endif
+
+#define PPM_FORMAT "P6\n"
+#define MAX_COLOR_VAL "255\n"
+
+typedef struct Color {
+    unsigned char r, g, b;
+} Color;
+
+typedef struct Pair {
+    int x, y;
+} Pair;
+
+enum { Rectx, Recty, w, h, Rectr, Rectg, Rectb };
+enum { Circlex, Circley, rad, Circler, Circleg, Circleb };
+
+void DrawRect(char* data, Pair* pixelsPos, Color* pixelsColor, int pixelCount, int imageSize[2]) {
+    int values[7];
+    char count = 0;
+    while (*data) {
+        if (*data >= '0' && *data <= '9') {
+            values[count] = strtoll(data, &data, 10);
+            count++;
+        }
+        else
+            data++;
+    }
+    for (int i = values[Recty]; i < min(values[Recty] + values[h], imageSize[1]); i++) {
+        for (int j = values[Rectx]; j < min(values[Rectx] + values[w], imageSize[0]); j++) {
+            pixelsColor[i * 800 + j].r = values[Rectr];
+            pixelsColor[i * 800 + j].g = values[Rectg];
+            pixelsColor[i * 800 + j].b = values[Rectb];
+        }
+    }
+}
+
+void DrawCircle(char* data, Pair* pixelsPos, Color* pixelsColor, int pixelCount, int imageSize[2]) {
+    int values[6];
+    char count = 0;
+    while (*data) {
+        if (*data >= '0' && *data <= '9') {
+            values[count] = strtoll(data, &data, 10);
+            count++;
+        }
+        else
+            data++;
+    }
+    for (int i = max(values[Circley] - values[rad], 0); i < min(values[Circley] + values[rad], imageSize[1]); i++) {
+        for (int j = max(values[Circlex] - values[rad], 0); j < min(values[Circlex] + values[rad], imageSize[0]); j++) {
+            if (((pixelsPos[i*800 + j].x - values[Circlex]) * (pixelsPos[i*800 + j].x - values[Circlex]))
+                + ((pixelsPos[i*800 + j].y - values[Circley]) * (pixelsPos[i*800 + j].y - values[Circley]))
+                <= values[rad] * values[rad]) {
+                pixelsColor[i*800 + j].r = values[Circler];

```

Comment:
I don't think we have fixed the size of the frame buffer to be 800x800, so this is probably not correct for other cases

Reply:
oh no! i forgot to change that